ogre/src/OgreLidarVisual.cc

```diff
@@ -0,0 +1,325 @@
+  {
+    ray->Clear();
+    ray.reset();
+  }
+
+  for (auto ray : this->dataPtr->deadZoneRayFans)
+  {
+    ray->Clear();
+    ray.reset();
+  }
+
+  this->ClearPoints();
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::Init()
+{
+  BaseLidarVisual::Init();
+  this->Create();
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::Create()
+{
+  this->ClearPoints();
+  this->dataPtr->receivedData = false;
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::ClearPoints()
+{
+  this->dataPtr->lidarPoints.clear();
+  this->dataPtr->receivedData = false;
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::ClearVisualData()
+{
+  this->dataPtr->noHitRayStrips.clear();
+  this->dataPtr->deadZoneRayFans.clear();
+  this->dataPtr->rayLines.clear();
+  this->dataPtr->rayStrips.clear();
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::SetPoints(const std::vector<double> &_points)
+{
+  this->dataPtr->lidarPoints = _points;
+  this->dataPtr->receivedData = true;
+}
+
+//////////////////////////////////////////////////
+void OgreLidarVisual::Update()
+{
+  if (!this->dataPtr->receivedData || this->dataPtr->lidarPoints.size() == 0)
+  {
+    ignwarn << "New lidar data not received. Exiting update function"
+            << std::endl;
+    return;
+  }
+
+  this->dataPtr->receivedData = false;
+  double horizontalAngle = this->minHorizontalAngle;
+  double verticalAngle = this->minVerticalAngle;
+
+  if (this->horizontalCount > 1)
+  {
+    this->horizontalAngleStep =
+        (this->maxHorizontalAngle - this->minHorizontalAngle) /
+              (this->horizontalCount - 1);
+  }
+  if (this->verticalCount > 1)
+  {
+    this->verticalAngleStep =
+        (this->maxVerticalAngle - this->minVerticalAngle) /
+              (this->verticalCount - 1);
+  }
+
+  if (this->dataPtr->lidarPoints.size() !=
+                  this->verticalCount * this->horizontalCount)
+  {
+    ignwarn << "Size of lidar data inconsistent with rays."
+            << " Exiting update function."
+            << std::endl;
+    return;
+  }
+
+  // Process each point from received data
+  // Every line segment, and every triangle is saved separately,
+  // as a pointer to a DynamicLine
+  for (unsigned int j = 0; j < this->verticalCount; ++j)
+  {
+    horizontalAngle = this->minHorizontalAngle;
+
+    if (j+1 > this->dataPtr->rayLines.size())
+    {
+      // Ray Strips fill in between the line areas that intersect an object
+      std::shared_ptr<OgreDynamicLines> line =
+                  std::shared_ptr<OgreDynamicLines>(
+                              new OgreDynamicLines(MT_TRIANGLE_STRIP));
+      #if (OGRE_VERSION <= ((1 << 16) | (10 << 8) | 7))
+            line->setMaterial("Lidar/BlueStrips");
+      #else
+            line->setMaterial(
+                this->Scene()->Material("Lidar/BlueStrips")->Clone());
```

Comment:
This returns an `ignition::rendering MateiralPtr` object but `setMaterial` expects an `Ogre::MateiralPtr` object. So windows build is still failing, see [console](https://build.osrfoundation.org/job/ign_rendering-pr-win/835/console) output

You'll need to get the material using ogre's material manager:

```
Ogre::MaterialPtr mat = Ogre::MaterialManager::getSingleton().getByName("Lidar/BlueStrips");
```